src/Extensions/S3Storage/S3StorageEgressProviderOptions.cs

```diff
@@ -14,38 +14,38 @@ internal sealed partial class S3StorageEgressProviderOptions
         [Display(
             ResourceType = typeof(OptionsDisplayStrings),
             Description = nameof(OptionsDisplayStrings.DisplayAttributeDescription_S3StorageEgressProviderOptions_Endpoint))]
-        public string Endpoint { get; set; }
+        public string? Endpoint { get; set; }
 
         [Display(
             ResourceType = typeof(OptionsDisplayStrings),
             Description = nameof(OptionsDisplayStrings.DisplayAttributeDescription_S3StorageEgressProviderOptions_BucketName))]
         [Required(AllowEmptyStrings = false)]
-        public string BucketName { get; set; }
+        public string? BucketName { get; set; }
```

Comment:
Since this configuration option is `[Required]` we know that it must have a value and not be null. 

The pattern I've been using elsewhere for required config options is to make them non-nullable with a default value of an empty string.

```suggestion
        public string BucketName { get; set; } = string.Empty;
```

Reply:
With changing this I suspect we'll also want to update a few other files to make `BucketName` non-nullable.